integration/maptest/map_test.go

```diff
@@ -0,0 +1,383 @@
+		index := incl.GetLeaf().GetIndex()
+		leafHash := incl.GetLeaf().GetLeafHash()
+		proof := incl.GetInclusion()
+
+		if got, want := leafHash, hasher.HashLeaf(treeID, index, hasher.BitLen(), leaf); !bytes.Equal(got, want) {
+			return fmt.Errorf("HashLeaf(%s): %x, want %x", leaf, got, want)
+		}
+		if err := merkle.VerifyMapInclusionProof(treeID, index,
+			leafHash, rootHash, proof, hasher); err != nil {
+			return fmt.Errorf("verifyMapInclusionProof(%x): %v", index, err)
+		}
+	}
+	return nil
+}
+
+// newTreeWithHasher is a test setup helper for creating new trees with a given hasher.
+func newTreeWithHasher(ctx context.Context, env *integration.MapEnv, hashStrategy trillian.HashStrategy) (*trillian.Tree, hashers.MapHasher, error) {
+	treeParams := stestonly.MapTree
+	treeParams.HashStrategy = hashStrategy
+	tree, err := env.AdminClient.CreateTree(ctx, &trillian.CreateTreeRequest{
+		Tree: treeParams,
+	})
+	if err != nil {
+		return nil, nil, nil
+	}
+
+	hasher, err := hashers.NewMapHasher(tree.HashStrategy)
+	if err != nil {
+		return nil, nil, nil
+	}
+	return tree, hasher, nil
+}
+
+func TestInclusion(t *testing.T) {
+	ctx := context.Background()
+	for _, tc := range []struct {
+		desc         string
+		HashStrategy trillian.HashStrategy
+		iv           [][]byte
+	}{
+		{
+			desc:         "maphasher single",
+			HashStrategy: trillian.HashStrategy_TEST_MAP_HASHER,
+			iv: [][]byte{
+				testonly.TransparentHash("A"), []byte("A"),
+			},
+		},
+		{
+			desc:         "maphasher multi",
+			HashStrategy: trillian.HashStrategy_TEST_MAP_HASHER,
+			iv: [][]byte{
+				testonly.TransparentHash("A"), []byte("A"),
+				testonly.TransparentHash("B"), []byte("B"),
+				testonly.TransparentHash("C"), []byte("C"),
+			},
+		},
+	} {
+		tree, hasher, err := newTreeWithHasher(ctx, env, tc.HashStrategy)
+		if err != nil {
+			t.Errorf("%v: newTreeWithHasher(%v): %v", tc.desc, tc.HashStrategy, err)
+			continue
+		}
+		pubKey, err := keys.NewFromPublicDER(tree.GetPublicKey().GetDer())
+		if err != nil {
+			t.Errorf("%v: NewFromPublicDER(%v): %v", tc.desc, tc.HashStrategy, err)
+			continue
+		}
+
+		leaves := createMapLeaves(tc.iv...)
+		if _, err := env.MapClient.SetLeaves(ctx, &trillian.SetMapLeavesRequest{
+			MapId:  tree.TreeId,
+			Leaves: leaves,
+			MapperData: &trillian.MapperMetadata{
+				HighestFullyCompletedSeq: 0xcafe,
+			},
+		}); err != nil {
+			t.Errorf("%v: SetLeaves(): %v", tc.desc, err)
+			continue
+		}
+
+		indexes := [][]byte{}
+		for _, l := range leaves {
+			indexes = append(indexes, l.Index)
+		}
+		getResp, err := env.MapClient.GetLeaves(ctx, &trillian.GetMapLeavesRequest{
+			MapId:    tree.TreeId,
+			Index:    indexes,
+			Revision: -1,
+		})
+		if err != nil {
+			t.Errorf("%v: GetLeaves(): %v", tc.desc, err)
+			continue
+		}
+
+		if err := verifyGetMapLeavesResponse(getResp, indexes, 1,
+			pubKey, hasher, tree.TreeId); err != nil {
+			t.Errorf("%v: verifyGetMapLeavesResponse(): %v", tc.desc, err)
```

Comment:
Add a `continue` here? I know it's the end of the loop now, but if you added another thing later on you'd get a surprising crash.